lib/matplotlib/tests/test_axes.py

```diff
@@ -6513,3 +6513,22 @@ def test_multiplot_autoscale():
     ax2.axhspan(-5, 5)
     xlim = ax1.get_xlim()
     assert np.allclose(xlim, [0.5, 4.5])
+
+
+@check_figures_equal(extensions=["pdf"])
+def test_2dcolor_plot(fig_test, fig_ref):
+    color = np.array([0.1, 0.2, 0.3])
+    # plot with 1D-color:
+    axs = fig_test.subplots(5)
+    axs[0].plot([1, 2], [1, 2], c=color.reshape((-1)))
+    axs[1].scatter([1, 2], [1, 2], c=color.reshape((-1)))
+    axs[2].step([1, 2], [1, 2], c=color.reshape((-1)))
+    axs[3].hist(np.arange(10), color=color.reshape((-1)))
+    axs[4].bar(np.arange(10), np.arange(10), color=color.reshape((-1)))
+    # plot with 2D-color:
+    axs = fig_ref.subplots(5)
```

Comment:
This is backwards, no? The 2D is the new thing to be tested, and the 1D is the reference.

Reply:
I agree with your assumption. 1D arrays have a high compatibility due to their similarity to tuples (a,b,c) and lists [a,b,c] while 2D arrays can be misunderstood. Therefore, 1D can be considered the reference.

However, despite the naming of the arguments, isn't this decorator performing a comparison statement (check equal), and therefore this is providing the choice between
`a == b`  and `b == a` ?

Reply:
Yes, it's just a matter of semantics, but the file names are different, the way it's presented on error is different, etc.

Reply:
I have changed the reference accordingly.